winter/core/state_key.py

```diff
@@ -0,0 +1,9 @@
+import typing
+
+import dataclasses
+
+
+@dataclasses.dataclass(frozen=True)
+class StateKey:
+    key: typing.Hashable
+    many: bool = False
```

Comment:
`many` field here limits the key-value storage convenience and possibilities. Please consider the following API, where key is a string (or any other hashable). So the user decides the structure of a value (not a key).
```
class QueryParameters:
    def __init__(self):
        self._mappings = {}

    def add(name: str, map_to: str = None):
        pass  # not implemented to shorter the example

query_params = SimpleComponent.simple_method.metadata.setdefault('query_params', QueryParameters())
query_params.add('param1')
query_params.add('param2')
```